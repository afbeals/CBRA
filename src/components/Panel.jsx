// Extenal
import React from 'react';
import PropTypes from 'prop-types';

// LOcal
import ErrorBoundary from '~Components/ErrorBoundary';

// Constants
const classname = 'panel';

// Component
const Panel = ({ title, children, className, render, borderless }) => (
  <ErrorBoundary>
    <div
      data-testid="panel"
      className={`${className ? `${className} ` : ''}${
        borderless ? `${borderless} ` : ''
      }${classname}`}
    >
      {typeof title === 'string' && (
        <h2 className={`${classname}__header`}>{title}</h2>
      )}
      {typeof title !== 'string' && (
        <div className={`${classname}__header`}>{title}</div>
      )}
      <div className={`${classname}__body`}>
        {children || render(`${className ? `${className}__body` : ''}`)}
      </div>
    </div>
  </ErrorBoundary>
);

Panel.propTypes = {
  title: PropTypes.node.isRequired,
  render: (props, propName, componentName) => {
    const { children, content, render } = props;
    if (!children && !content && !render) {
      return new Error(
        `'${propName}'. Must provide either content prop, children, or render prop in ${componentName}.`,
      );
    }
    return null;
  },
  children: (props, propName, componentName) => {
    const { children, content, render } = props;
    if (!children && !content && !render) {
      return new Error(
        `'${propName}'. Must provide either content prop, children, or render prop in ${componentName}.`,
      );
    }
    return null;
  },
  borderless: PropTypes.bool,
  className: PropTypes.string,
};

export default Panel;
