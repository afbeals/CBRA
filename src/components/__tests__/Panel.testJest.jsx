// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import Panel from '../Panel';

// tests
describe('Panel', () => {
  afterEach(cleanup);
  it('should render Panel', () => {
    const { getByTestId } = render(
      <Panel title={'panel'} className={'custom'}>
        children
      </Panel>,
    );
    expect(getByTestId('panel')).toMatchSnapshot();
  });
});
