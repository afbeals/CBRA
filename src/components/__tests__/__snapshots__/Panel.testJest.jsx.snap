// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render Panel 1`] = `
<div
  class="custom panel"
  data-testid="panel"
>
  <h2
    class="panel__header"
  >
    panel
  </h2>
  <div
    class="panel__body"
  >
    children
  </div>
</div>
`;
